### PR TITLE
Issue #2080 Added RHEL and CenOS to CLI completion usage

### LINKED ISSUE #2080: Update tab completion UX and help doc for RHEL and centos
### General information

  * Minishift version: 
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

### Steps to reproduce

  1. $ ./minishift completion --help
```
       Outputs minishift shell completion for the given shell (bash or zsh)
	This depends on the bash-completion binary.  Example installation instructions:
	OS X:
		$ brew install bash-completion
		$ source $(brew --prefix)/etc/bash_completion
		$ minishift completion bash > ~/.minishift-completion  # for bash users
		$ minishift completion zsh > ~/.minishift-completion  # for zsh users
		$ source ~/.minishift-completion
	Fedora:
		$ dnf install bash-completion
		$ minishift completion bash > ~/.minishift-completion  # for bash users
		$ source ~/.minishift-completion
        [...]
```

### Expected
$ minishift completion  --help
```
       Outputs minishift shell completion for the given shell (bash or zsh)
	This depends on the bash-completion binary.  Example installation instructions:
	OS X:
	    Step 1 $ brew install bash-completion
	    Step 2 $ source $(brew --prefix)/etc/bash_completion
	    Step 3 $ minishift completion bash > ~/.minishift-completion  # for bash users
	    Step 4 $ minishift completion zsh > ~/.minishift-completion  # for zsh users
	    Step 5 $ source ~/.minishift-completion
	Linux:
	  Fedora: 
            Step 1 $ dnf install bash-completion
          RHEL:
            Step 1 yum install bash-completion

	    Step 2 $ minishift completion bash > ~/.minishift-completion  # for bash users
	    Step 3 $ source ~/.minishift-completion
            [...]
```
We can add bash completion uses for other linux flavor such as ```CentOS``` 
### Actual
The actual output is inclined towards a specific flavor of linux ```Fedora``` where as we have other major flavor like RHEL, CentOS (Mostly ```m start``` tested on these flavors.) can be comply minishift tab completion. See the doc https://docs.openshift.org/latest/minishift/command-ref/minishift_completion.html

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Fixes #2080 

Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
@minishift/minishift-dev @amitkrout @robin-owen does it look fine to you?

File: cmd/minishift/cmd/completion.go [hunk 1]
```diff
@@ -32,15 +32,16 @@ const longDescription = `
...
-		$ brew install bash-completion
-		$ source $(brew --prefix)/etc/bash_completion
-		$ minishift completion bash > ~/.minishift-completion  # for bash users
-		$ minishift completion zsh > ~/.minishift-completion  # for zsh users
-		$ source ~/.minishift-completion
-	Fedora:
-		$ dnf install bash-completion
-		$ minishift completion bash > ~/.minishift-completion  # for bash users
-		$ source ~/.minishift-completion
+			$ brew install bash-completion
```
Comment (budhrg) [inline cmd/minishift/cmd/completion.go]: This seems to be lot of extra spaces.

I think for better readability we can write as: 

```
This depends on the bash-completion binary.  Example installation instructions:

OS X:

    $ brew install bash-completion
    $ source $(brew --prefix)/etc/bash_completion
    $ minishift completion bash > ~/.minishift-completion  # for bash users
    $ minishift completion zsh > ~/.minishift-completion  # for zsh users
    $ source ~/.minishift-completion

RHEL/CentOS/Fedora:

    $ yum install bash-completion # for RHEL/CentOS
    $ dnf install bash-completion   # for Fedora
    $ minishift completion bash > ~/.minishift-completion  # for bash users
    $ source ~/.minishift-completion
```

Comment (amitkrout) [reply cmd/minishift/cmd/completion.go]: Single tab would be file for steps indentation

File: cmd/minishift/cmd/completion.go [hunk 2]
```diff
@@ -31,15 +31,20 @@ const longDescription = `
 	Outputs minishift shell completion for the given shell (bash or zsh)
 
 	This depends on the bash-completion binary.  Example installation instructions:
+
 	OS X:
```
Comment (robin-owen) [inline cmd/minishift/cmd/completion.go]: This should refer to macOS instead of OS X.

File: cmd/minishift/cmd/completion.go [hunk 3]
```diff
@@ -31,15 +31,20 @@ const longDescription = `
...
 		$ source ~/.minishift-completion
-	Fedora:
-		$ dnf install bash-completion
-		$ minishift completion bash > ~/.minishift-completion  # for bash users
+
+	RHEL/Fedora:
+
+		$ yum install bash-completion				# for RHEL
+		$ dnf install bash-completion   			# for Fedora
+		$ minishift completion bash > ~/.minishift-completion 	# for bash users
```
Comment (robin-owen) [inline cmd/minishift/cmd/completion.go]: The instructions for both `bash` and `zsh` were included for macOS. Is there a reason that only `bash` is included here?

Comment (LalatenduMohanty) [reply cmd/minishift/cmd/completion.go]: @robin-owen yes, because bash is default in Linux.

File: cmd/minishift/cmd/completion.go (hunk 2 above)
Comment (LalatenduMohanty) [reply cmd/minishift/cmd/completion.go]: @robin-owen sure, I was expecting this comment.

File: cmd/minishift/cmd/completion.go (hunk 3 above)
Comment (robin-owen) [reply cmd/minishift/cmd/completion.go]: @LalatenduMohanty I'm aware, but may be biased, as I use zsh. ;)

Comment (LalatenduMohanty) [reply cmd/minishift/cmd/completion.go]: @robin-owen I never used zsh, but if you steps which I can add here, we should be able to add the information.

Comment (robin-owen) [reply cmd/minishift/cmd/completion.go]: @LalatenduMohanty I just verified that the steps used for macOS work properly on Fedora 27 using zsh. Specifically:

```sh
$ minishift completion zsh > ~/.minishift-completion
$ source ~/.minishift-completion
```

Comment (LalatenduMohanty) [reply cmd/minishift/cmd/completion.go]: @robin-owen Added zsh step. Please check.

**Pull request merged**